internal/cliwrapper/cliwrapper_test.go

```diff
@@ -50,58 +50,71 @@ func Podman_ImageExists(t *testing.T, connectionName string) {
 }
 
 func TestPodman_ImageExists(t *testing.T) {
-	Podman_ImageExists(t, "")
+	podmanImageExists(t, nil)
 }
 
 func TestPodman_Remote_ImageExists(t *testing.T) {
-	var tmpPodmanSocketCmd *exec.Cmd
-
-	// Check if there is an existing connection of `podman-machine-default` since this is included when installing
-	// podman desktop for macOS.
+	// Check if there is an existing connection of `podman-machine-default`
+	// since this is included when installing podman desktop for macOS.
 	connectionName := "podman-machine-default"
-	chkDefaultConnectionCmd := exec.Command(tests.GetPodmanPath(), "--connection", connectionName, "system", "info") //nolint:gosec
-	if err := chkDefaultConnectionCmd.Run(); err == nil {
-		Podman_ImageExists(t, connectionName)
-	} else if runtime.GOOS == "linux" {
-		// The podman-machine-default doesn't exist then for Linux, create a temporary socket
-
-		// Setup
-		connectionName = "arcaflow-engine-deployer-podman-test"
-		podmanSocketPath := "unix:///var/tmp/" + connectionName + ".sock"
-
-		tmpPodmanSocketCmd = exec.Command(tests.GetPodmanPath(), "system", "service", "--time=0", podmanSocketPath)
-		if err := tmpPodmanSocketCmd.Start(); err != nil {
-			t.Fatalf("Failed to create temporary podman socket")
+	chkDefaultConnectionCmd := exec.Command(tests.GetPodmanPath(), "--connection", connectionName, "system", "info") //nolint:gosec  // Command line is trusted
+	if err := chkDefaultConnectionCmd.Run(); err != nil {
+		// The podman-machine-default connection doesn't exist, so try to create
+		// an alternative connection service.  For now, only try this on Linux.
+		//
+		//goland:noinspection GoBoolExpressions  // The linter cannot tell that this expression is not constant.
+		if runtime.GOOS != "linux" {
+			t.Skipf("There is no default Podman connection and no support for creating it on %s.", runtime.GOOS)
 		}
 
-		addConnectionCmd := exec.Command(tests.GetPodmanPath(), "system", "connection", "add", connectionName, podmanSocketPath)
-		if err := addConnectionCmd.Run(); err != nil {
-			t.Fatalf("Failed to add connection: " + connectionName)
-		}
+		connectionName = createPodmanConnection(t)
+	}
+
+	// Run the test
+	podmanImageExists(t, &connectionName)
+}
 
-		// Run test
-		Podman_ImageExists(t, connectionName)
+// createPodmanConnection creates a Podman API service process and configures
+// a Podman "connection" to allow it to be used for remote Podman invocations.
+func createPodmanConnection(t *testing.T) (connectionName string) {
+	// Setup
+	t.Logf("Adding a local Podman API service and connection.")
+	connectionName = "arcaflow-engine-deployer-podman-test"
```

Comment:
This seems like a great opportunity to randomize the name to avoid races just in case it's ever possible to get two instances running on the same system ...

Reply:
Yeah....  A little experimentation suggests that the "connection" list is persistent and shared between logins of a single user.  So, potentially we should be randomizing both the socket file name _and_ the service connection name, in order to avoid conflicts on either one.

> How are GitHub Actions isolated?

Pretty well:  actions from different branches and multiple actions from a single branch can all be run concurrently.  For the most part, if you want to have shared/persistent context, you have to send it to and restore it from a cache (it's like a tarball backup/restore).  I think it's done via VMs.

So, I think the foremost concern would be portions of a single test run interfering with other portions run in parallel.  But, I s'pose, we could get in trouble in testing scenarios other than the CI.

Reply:
OK, @dbutenhof, we now randomize both the connection name and the socket path.  😁

Reply:
Interesting ... I can't Resolve my comment on this branch ...

Reply:
This PR is in Ed's repo...so, you're not a "maintainer" here.  🙂

However, apparently I, perhaps because I'm the author of this PR, _can_...so, done.

Reply:
Exactly as I expected. 😆